src/upstream_patches_ui.sh

```diff
@@ -276,13 +282,19 @@ function show_series_details()
       for option in "${selected_options[@]}"; do
         case "$option" in
           'Bookmark')
-            create_loading_screen_notification "Bookmarking patch(es)"$'\n'"- ${patch_title}"
-            download_series "$total_patches" "$patch_url" "${lore_config['download_to']}" "$patch_title"
-            add_series_to_bookmark "${target_patch}" "${lore_config['download_to']}"
+            create_loading_screen_notification 'Bookmarking patch(es)'$'\n'"- ${patch_title}"
+            {
+              download_series "$total_patches" "$patch_url" "${lore_config['download_to']}" "$patch_title" &&
+                add_series_to_bookmark "${target_patch}" "${lore_config['download_to']}"
+            } || {
+              create_message_box 'Error' 'Could not bookmark patch(es)'$'\n'"- ${patch_title}"
+            }
```

Comment:
Please ignore my comments; now I can see that you are checking the return values. However, could you use a simple `if` condition here? It should be more readable and aligned with the code style.